### PR TITLE
TheOdinProject/curriculum: Modify and add headings and section placement

### LINKED ISSUE #24342: Style Guide - Foundations/JavaScript Basics 2: Headings and Section Placement
# Foundations/JavaScript Basics 2:

This issue is part of the Style Guide Community Project, a way of offering members of The Odin Project community a chance to contribute to our curriculum. Users who wish to take part and contribute to any Style Guide issues should have thoroughly read the [layout style guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md) and [The Odin Project contributing guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md).
  
## Description of Issue

This issue ensures that headings are properly formatted and that all sections are placed in the correct order.

Each applicable lesson for this issue should meet the following bulleted requirements:

- The main headings use the Markdown `###` syntax.
- Sub-headings that are not inline use the Markdown `####` syntax.
- Sub-headings that are inline use the Markdown `**sub-heading**` (i.e. bold) syntax.
- Both headings and sub-headings use Wikipedia style Title Case.
- There is a newline before and after headings and non-inline sub-headings.
- The order of existing sections follows the below format:
	- Introduction
	- Lesson Overview
	- Custom Sections (lessons) / Pre-Assignment Sections (projects)
	- Assignment
	- Practice (lessons) / Post-Assignment Sections (projects)
	- Knowledge Check
	- Additional Resources
- Additionally, there should be an empty newline at the end of the file.
 
## Acceptance Criteria

- [ ] The following lessons are updated according to the issue description above:
	- [ ] [Clean Code](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/clean-code)
            - The opening "Writing Clean Code" heading should be renamed to "Introduction"
	- [ ] [Installing Node.js](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/installing-node-js)
	- [ ] [DOM Manipulation and Events](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/dom-manipulation-and-events)
	- [ ] [Fundamentals Part 5](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/fundamentals-part-5)
	- [ ] [Project: Calculator](https://www.theodinproject.com/lessons/foundations-calculator)

### PR DESCRIPTION
**Pull request by c-yip**
Complete the following REQUIRED checkboxes:

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`

Complete the following checkboxes ONLY IF they are applicable to your PR. You can complete them later if they are not currently applicable:
-   [x] I have previewed all lesson files included in this PR with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure the Markdown content is formatted correctly
-   [x] I have ensured all lesson files included in this PR follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

<hr>

**1. Because:**

Closes #24342

**2. This PR:**

- Clean Code
    - Reorder headings
- Installing Node.js
    - Edit headings to conform with Wikipedia title case
    - Add Lesson Overview heading with content.
- DOM Manipulation and Events
    - Edit headings to conform with Wikipedia title case
- Fundamentals Part 5
    - Add Introduction heading
- Project: Calculator
    - Add pre-assignment and post-assignment headings

**3. Additional Information:**

I did not add content to many of the headings I added. All headings that were added were added to conform with the style guide.

### CODE REVIEW

File: foundations/javascript_basics/project_calculator.md [hunk 1]
```diff
@@ -46,3 +48,5 @@ Here are some use cases (abilities your project needs to have):
 * Add a "backspace" button, so the user can undo if they click the wrong number.
 * Add keyboard support!
 </div>
+
+### POST-ASSIGNMENT SECTION HEADING
```
Comment (zachmmeyer) [inline foundations/javascript_basics/project_calculator.md:52]:
```suggestion

```
I don't believe this is necessary. It also would just say `POST-ASSIGNMENT SECTION HEADING` which wouldn't look very good. You're supposed to write something else, but I don't think anything would go here.

**Review by zachmmeyer — CHANGES_REQUESTED** [review]
Everything looks pretty good. I had one direct change request, and you're missing the newlines at the end of the files.

**Comment by zachmmeyer** [comment]
@c-yip Are you able to resolve the merge conflicts?

**Comment by c-yip** [comment]
@zachmmeyer I will resolve the merge conflicts right now. 

Regarding the newlines at the end of the files, my local repo shows newlines at all the files I made commits to. I am not sure why the newlines do not show up on GitHub. Perhaps GitHub does not show the newlines at the end of files in its code preview?

**Comment by zachmmeyer** [comment]
> @zachmmeyer I will resolve the merge conflicts right now. 
> 
> Regarding the newlines at the end of the files, my local repo shows newlines at all the files I made commits to. I am not sure why the newlines do not show up on GitHub. Perhaps GitHub does not show the newlines at the end of files in its code preview?

That's OK, do not worry about the newlines then.

**Comment by c-yip** [comment]
@zachmmeyer I resolved the merge conflicts. Please let me know if there are any other issues.

**Review by zachmmeyer — APPROVED** [review]
Looks good to me! Thank you!

**Pull request merged**